### PR TITLE
Add AWS creds to build-deploy-pudl action and copy outputs to s3 bucket

### PR DESCRIPTION
**Pull request by bendnorman**
This PR adds support for copying nightly build outputs to our new `s3://intake.catalyst.coop` bucket. 

See the catalyst-cooperative/pudl-catalog#70 PR for the supporting changes in the `pudl-catalog` repo.

## PR Checklist

Before requesting a review of your pull request, please make sure you've done the
following:

* [ ] Merge the most recent version of `dev` (or the appropriate upstream branch) into
      your branch and resolved any merge conflicts. You may need to do this several
      times over the course of a PR as `dev` changes frequently.
* [ ] Verify that all of the CI checks on your PR are passing. See
      [Running Tests with Tox](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
      for details on how to run the full test suite locally if you need to debug a
      particular failure.
* [ ] Ensure that the docstrings for any new modules, classes, functions, or methods are
      descriptive enough for developers and users to understand your code.
* [ ] If you expanded data coverage or changed the outputs, ensure that the full
      [data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
      pass locally on a fresh DB.
* [ ] If you've added new functions or classes, ensure that they have at least basic
      unit tests.
* [ ] If you've added new analyses, make sure they include defensive sanity checks that
      will catch unexpected data issues.
* [ ] Update the
      [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html)
      to reflect your changes. Make sure to reference the PR and any related issues.
* [ ] Do your own review of the PR. Add comments highlighting areas where you have
      questions you'd like reviewers to answer, known issues, solutions you're
      unsatisfied with, or other things that deserve special attention from the
      reviewer.

### CODE REVIEW

File: docker/gcp_pudl_etl.sh [hunk 1]
```diff
@@ -62,9 +62,13 @@ function shutdown_vm() {
...
 function copy_outputs_to_intake_bucket() {
-    echo "Copying outputs to intake bucket"
+    echo "Copying outputs to GCP intake bucket"
     gsutil -m -u $GCP_BILLING_PROJECT cp -r "$CONTAINER_PUDL_OUT/sqlite/*" "gs://intake.catalyst.coop/$GITHUB_REF"
     gsutil -m -u $GCP_BILLING_PROJECT cp -r "$CONTAINER_PUDL_OUT/parquet/epacems/*" "gs://intake.catalyst.coop/$GITHUB_REF"
+
+    echo "Copying outputs to AWS intake bucket"
+    aws s3 cp "$CONTAINER_PUDL_OUT/sqlite" "s3://intake.catalyst.coop/$GITHUB_REF" --recursive
+    aws s3 cp "$CONTAINER_PUDL_OUT/parquet/epacems" "s3://intake.catalyst.coop/$GITHUB_REF" --recursive
 }
```
Comment (zaneselvans) [inline docker/gcp_pudl_etl.sh:72]: I would really like to change our local file layout to be flat, as it is in the buckets, for simplicity and so they can both be treated the same. Beyond this PR but just FYI...

**Review by zaneselvans — APPROVED** [review]

File: docker/gcp_pudl_etl.sh (hunk 1 above)
Comment (bendnorman) [reply docker/gcp_pudl_etl.sh:72]: Oh so you'd like all of the ETL's outputs to live in one flat directory?

**Comment by bendnorman** [comment]
I need to add an explanation on how to download the nightly build outputs using the AWS cli before I merge this in.

File: docker/gcp_pudl_etl.sh (hunk 1 above)
Comment (zaneselvans) [reply docker/gcp_pudl_etl.sh:72]: I think so. They all have unique names, and we've started to comingle metadata outputs (JSON) with SQLite DB outputs. I just don't know that it adds anything to divide them up by filetype, and it does mean there's more path management to do.

**Comment by bendnorman** [comment]
I added the aws and gcs set up to the nightly builds docs so this PR is ready to go once the CI passes.

**Pull request merged**